kythe/go/platform/tools/extraction/extractrepo/extractrepo.go

```diff
@@ -89,11 +92,44 @@ func verifyFlags() {
 		log.Println("You must provide a non-empty -output")
 	}
 
+	if *tempRepoDir != "" {
+		if !empty(*tempRepoDir) {
+			hasError = true
+			log.Println("-tmp_repo_dir must be an empty directory.")
+		}
+	}
+
 	if hasError {
 		os.Exit(1)
 	}
 }
 
+func empty(dir string) bool {
+	stat, err := os.Stat(dir)
+	if os.IsNotExist(err) {
+		return os.MkdirAll(dir, 0755) == nil
+	}
+	if err != nil {
+		log.Printf("Unexpected stat problem: %v", err)
+		return false
+	}
+	if !stat.IsDir() {
+		return false
+	}
+	f, err := os.Open(dir)
```

Comment:
I'm not sure this optimization really helps. This could more simply be

```
fi, err := ioutil.ReadDir(dir)
if os.IsNotExist(err) {
   return os.MkdirAll(dir, 0755) == nil
} else if err != nil {
   log.Printf("Error checking directory: %v", err)
}
return err == nil && len(fi) == 0
```

I realize we might wind up reading more than we strictly need to, but it's just a one-time startup cost.

Reply:
Ah yea fair enough.  Especially since this is intended to not live long anyways, I probably got a bit carried away.